pool.go

```diff
@@ -298,15 +368,18 @@ func (p *pool) doFetch(ctx context.Context, from string, c cid.Cid) (b blocks.Bl
 		}
 	}
 
-	resp, err := p.config.Client.Do(req)
+	resp, err := p.config.SaturnClient.Do(req)
 	if err != nil {
 		return nil, err
 	}
+	defer resp.Body.Close()
+
 	fb = time.Now()
 	code = resp.StatusCode
 	proto = resp.Proto
 	respReq = resp.Request
-	defer resp.Body.Close()
+
+	// TODO: What if the Saturn node is malicious ? We should have an upper bound on how many bytes we read here.
 	rb, err := io.ReadAll(resp.Body)
```

Comment:
@willscott @lidel Is this a concern ?

Reply:
If L1 returned HTTP Status Code 200 but the hash does not match, it should be penalized  way more than if it returned with error (code other than 200) or timeouted.

Reply:
PR update with both these changes.